File: expressdine-client/src/Services/ComplaintService.js

import axios from "axios";
const ed_url = "http://localhost:8080/complaint/";

export class ComplaintService{
  getComplaintByRestOwner = (ownerId) => {
    return axios.get(ed_url + "getByRestOwner/" + ownerId);
  };

  addComplaint = (complaint) => {
    alert(complaint);
    return axios.post(ed_url + "add", complaint);
  };

  getComplaintsByUser = (userId) => {
    return axios.get(ed_url + "getByUser/" + userId);
  };

  resolveComplaint = (complaintId) => {
    return axios.put(ed_url + "resolve/"+ complaintId);
  };
}

export default new ComplaintService();

File: ExpressDine_Server/src/main/java/com/cybage/service/ComplaintService.java
package com.cybage.service;

import java.util.List;

import com.cybage.model.Complaint;

public interface ComplaintService {
	
	void addComplaint(Complaint complaint);

	List<Complaint> getComplaintByRestOwnerId(int id);

	void resolveComplaint(int id);

	List<Complaint> getComplaintsByUser(int id);

}
